tests/test_apply.py

```diff
@@ -68,6 +68,10 @@ class TestStateApply:
     def test_qubit_state_vector(self, init_state, device, tol):
         """Test PauliX application"""
         dev = device(1)
+
+        if dev.backend_name == "unitary_simulator":
+            pytest.skip("Test only runs for backends that are not the unitary simulator.")
```

Comment:
Looks good! For maintainablity, I would suggest moving this into a fixture (e.g., `skip_unitary`), and adding it to function arguments where it should be skipped.